server/clientRoutes.js

```diff
@@ -0,0 +1,100 @@
+const express = require('express');
+const assert = require('bsert');
+const Config = require('bcfg');
+const logger = require('./logger');
+
+function clientsRouter(clients) {
+  const router = express.Router({ mergeParams: true });
+  let token, id, config, reqClients;
+
+  // return object with
+  // relevant info for each client
+  // keyed to the client id
+  const clientInfo = {};
+  clients.forEach(
+    client =>
+      (clientInfo[client.config.str('id')] = {
+        chain: client.config.str('chain', 'bitcoin')
+      })
+  );
+  router.get('/', (req, res) => res.status(200).json(clientInfo));
+
+  // middleware for setting constants based on
+  // the route being used
+  router.use('/:id', (req, res, next) => {
+    id = req.params.id;
+    if (!clients.has(id))
+      return res
+        .status(404)
+        .send(`Sorry, there was no client with the id ${id}`);
+    const clientObj = clients.get(id);
+    config = clientObj.config;
+    assert(config instanceof Config, 'client needs bcfg config');
+
+    // store clients to be used in the routes
+    // handled in the next middleware
+    reqClients = {
+      node: clientObj.nodeClient,
+      wallet: clientObj.walletClient,
+      multisig: clientObj.multisigClient
+    };
+    next();
+  });
+
+  // all routes for a given client
+  // the client param dictates client, node,
+  // wallet, or multisig, will be used
+  router.use('/:id/:client/', async (req, res) => {
+    const { method, path, body, query } = req;
```

Comment:
Curious as to why you didn't destructure `params` like the other `req` properties

Reply:
What is to stop someone from selecting a client that they shouldn't have access to? Maybe the `Proxy` api could be useful here, or a `Class` with a custom getter. This may be out of scope of this PR

Reply:
Right now, the assumption with the current setup is that there will be no restrictions on which clients can connect to which nodes. This has been the ongoing assumption for beta. But yeah, that's basically why this has been built using the proxy where right now it's not really necessary, in the future though the proxy lets us add a per-node authentication layer.

Reply:
I'll add destructuring for the params. Good call 👍